server/routes/api/cron/schema.ts

```diff
@@ -0,0 +1,44 @@
+import crypto from "crypto";
+import { isEmpty } from "lodash";
+import { z } from "zod";
+import env from "@server/env";
+import BaseSchema from "../BaseSchema";
+
+export const CronSchema = BaseSchema.extend({
+  body: z.object({
+    token: z.string().optional(),
+    limit: z.number().gt(0).default(500),
+  }),
+  query: z.object({
+    token: z.string().optional(),
+    limit: z.number().gt(0).default(500),
+  }),
+}).superRefine((req, ctx) => {
+  if (isEmpty(req.body.token) && isEmpty(req.query.token)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "token is required",
+      fatal: true,
+    });
+
+    return z.NEVER;
+  }
+
+  const token = (req.body.token ?? req.query.token) as string;
```

Comment:
Interesting, not sure how I feel about having the authentication inside data validation, feels like this is mixing concerns?

Reply:
You sound quite right. Perhaps we can retain `token` presence and `token.length` checks here and move the `crypto` checks inside the API handler?

Reply:
Separated concerns! Kept `token.length` check _with_ `crypto` check, made sense _semantically_ :+1: